pkg/model/series.go

```diff
@@ -7,26 +7,59 @@ import (
 	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
 )
 
+const (
+	AverageAggregationType    = "avg"
+	sumAggregationType        = "sum"
+	firstValueAggregationType = "first"
```

Comment:
I think first is confusing. But I understand why it's there.

Normally in prometheus land `{app="foo"}` returns an instant vector where there will be only one point per unique series per timestamp. This is done using looback period.

Check this one out https://www.linkedin.com/posts/julius-volz_promql-data-selection-explained-selectors-activity-7066421381151768578-Svsm/?trk=public_profile_like_view

So technically `first({app="foo"}) by (pod)` should never be needed, but it seems currently we actually returns a range vector for `{app="foo"}`.


Currently this is a bit invisible because we're doing `sum(sum_over_time({app="foo"}[step])) by (pod)`

I don't think it's super important for now, but if we ever want to have a query lang we most likely should get this resolved.

Reply:
I agree. Also, wording the current and new behavior is very nuanced. `{app="foo"}` in Pyroscope doesn't produce a vector at all (instant or range) but a single value at the evaluation time. It does this by first aggregating across series (with "sum" or "first" to avoid dealing with a range vector) and then across time (`sum_over_time(...[step])`).

I would be happy to remove the first aggregation completely. This would let duplicate timestamps in the intermediary result which `sum_over_time` (or `avg_over_time`) will take care off anyway.